low-level/solution/lr2/server/manifest.py

```diff
@@ -0,0 +1,348 @@
+            server_hw_data = self.get_server_hw_info()
+            info = self.get_server_info(server_hw_data)
+            resource_found = True
+        elif leaf_node == "hw":
+            server_hw_data = self.get_server_hw_info()
+            info = self.get_hw_manifest_info(server_hw_data, "hw")["hw"]
+            resource_found = True
+        elif leaf_node in ["sw", "fw"]:
+            for resource, method in self.server_resources[leaf_node].items():
+                try:
+                    info.update({resource: method()})
+                    resource_found = True
+                except Exception as err:
+                    logger.error(
+                        self.log.svc_log(f"{err.__class__.__name__}: {err}"))
+                    info = None
+        else:
+            server_hw_data = self.get_server_hw_info()
+            for node in nodes:
+                resource, _ = ServerResourceMap.get_node_details(node)
+                for res_type in self.server_resources:
+                    method = self.server_resources[res_type].get(resource)
+                    if not method:
+                        logger.error(
+                            self.log.svc_log(
+                                f"No mapping function found for {res_type}"))
+                        continue
+                    try:
+                        if res_type == "hw":
+                            info = method(server_hw_data, resource)
+                        else:
+                            info = method()
+                        resource_found = True
+                    except Exception as err:
+                        logger.error(
+                            self.log.svc_log(f"{err.__class__.__name__}: {err}"))
+                        info = None
+                if resource_found:
+                    break
+
+        if not resource_found:
+            msg = f"Invalid rpath or manifest provider doesn't have support for'{rpath}'."
+            logger.error(self.log.svc_log(f"{msg}"))
+            raise ResourceMapError(errno.EINVAL, msg)
+
+        return info
+
+    def get_server_info(self, server_hw_data):
+        """Get server manifest information."""
+        server = []
+        info = {}
+        for res_type in self.server_resources:
+            info.update({res_type: {}})
+            for fru, method in self.server_resources[res_type].items():
+                try:
+                    if res_type == "hw":
+                        info[res_type].update({fru: method(server_hw_data, fru)})
+                    else:
+                        info[res_type].update({fru: method()})
+                except Exception as err:
+                    logger.error(
+                        self.log.svc_log(f"{err.__class__.__name__}:{err}"))
+                    info[res_type].update({fru: []})
+        info["last_updated"] = int(time.time())
+        server.append(info)
+        return server
+
+    def get_server_hw_info(self):
```

Comment:
get_server_hw_info() - is giving manifest o/p file, how it is different from get_hw_manifest_info() ? Add enough description for each of the functions around, quite confusing

Reply:
Done

Reply:
get_server_hw_info() => It will return hole lshw filtered data
"""Get server hw information."""

get_hw_resources_info() ==> It will return individual hw resource information. (e.g. disk, system etc.)
"""Get server hw resource information."""